Assets/MixedRealityToolkit/Graph/Plugins/WSA/Microsoft.Identity.Client.dll.meta

```diff
@@ -0,0 +1,114 @@
+fileFormatVersion: 2
+guid: cbf1e91417ffe6c46b7e1b1e444ef8a7
```

Comment:
Where does this dll come from?

Is there source code that goes along with this?

Reply:
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet

Reply:
I wonder if there is a nice way to reference this dependency via GitHub.

My only concern here is if people already use a library in their projects and they get conflicts.

Reply:
@StephenHodgson I also hate the fact that Unity does not have a nice way to reference NuGet packages, or am I wrong about that? 

Options I can think of: we make this DLL disabled by default in Unity, so it does not cause any conflicts whatsoever, until users explicitly enable it. Or we don't even include the DLL and we mention in the documentation that developers require MSAL DLL to use MsalGraphAuthentication.cs, provided with the SDK?

Reply:
We could add a auto downloader if they so choose to enable these features

Reply:
It would just grab the latest from the github releases

Reply:
@StephenHodgson how do I hook to the auto downloader?

Reply:
We would have to write something in Unity that checks if they already have the dll in their project and then grab it from a direct link on github.  Or we could use a submodule approach and take it in directly.

Reply:
Got it. We really need a system to reference NuGets from Unity. The change you suggest makes totally sense to me but I don't have the cycles to do it. Can we leave the DLL embedded for now and create a separate work item to create a system that allows managing NuGets that everyone can benefit and hook in? Honestly, I hate the fact that Unity does not integrate with NuGet yet.

By the way, I stumbled up on https://github.com/GlitchEnzo/NuGetForUnity. Might be interesting to take a look.